rest/handler.go

```diff
@@ -132,17 +132,17 @@ func (h *handler) invoke(method handlerMethod) error {
 		}
 	}
 
-	h.logRequestLine()
-
 	switch h.rq.Header.Get("Content-Encoding") {
 	case "":
 		h.requestBody = h.rq.Body
 	case "gzip":
 		if h.requestBody, err = gzip.NewReader(h.rq.Body); err != nil {
+			h.logRequestLine()
```

Comment:
This approach works, but seems prone to future regressions if we add new error exits from invoke prior to logRequestLine being run.

You could avoid this by defining a bool to track whether logRequestLine has been called, and then add a defer around line 134 that checks that bool and logs the request line if it hasn't already been logged.

Reply:
Make sense, defined a new bool isRequestLogged to track whether logRequestLine has been called, and included a defer around line 134 that checks that bool and logs the request line if it hasn't already been logged.